dashboards-observability/public/components/custom_panels/helpers/utils.tsx

```diff
@@ -297,9 +297,9 @@ export const displayVisualization = (metaData: any, data: any, type: string) =>
       visualizations={getVizContainerProps({
         vizId: type,
         rawVizData: data,
-        query: {},
+        query: { rawQuery: metaData.query === undefined ? {} : metaData.query },
```

Comment:
In query parameter Why are we checking `metaData.query === undefined`? The statement will never be reached while being true. Check: https://github.com/opensearch-project/observability/blob/d243e918aff872efe7fed3769662ebddfec56b2e/dashboards-observability/public/components/custom_panels/helpers/utils.tsx#L292-L294

You may simply change this to:
```
query: { rawQuery: metaData.query }
```

Reply:
@ps48 we do have a test case where, we are checking with empty visualization, so i had to fix it with this. https://github.com/opensearch-project/observability/blob/e4a199be779eac368d0682fc636ee8d12cbaa05b/dashboards-observability/public/components/custom_panels/helpers/__tests__/utils.test.tsx#L128

Please let me know if we have to remove the test case or update the test case with valid visualization data.